examples/fishnet_api/main.py

```diff
@@ -102,16 +118,58 @@ async def get_executions() -> List[Execution]:
     return await Execution.fetch_all()
 
 
+@app.get("/executions/{dataset_ID}")
+async def get_executions_by_dataset(dataset_ID: str) -> List[Execution]:
+    return await Execution.query(datasetID=dataset_ID)
+
+
 @app.get("/user/{address}/executions")
 async def get_user_executions(address: str) -> List[Execution]:
     return await Execution.query(owner=address)
 
 
+@app.get("/Permission")
+async def permission():
+    status = []
+    permission = await Permission.fetch_all()
```

Comment:
That is not what we need. We need to target all datasets and allow for pagination so that we do not fetch all datasets (which takes too much time).

We then check the user's permissions on each dataset and return their permission status for that dataset together with that list of datasets.